tests/test_plotly_versions.py

```diff
@@ -0,0 +1,129 @@
+import os
+
+# Import the utility function for creating test data
+import time
+
+import psycopg2
+import requests
+
+from .test_livedata import generate_key
+from .utils.db import add_instrument_data_run, set_reduction_request_queue
+
+LIVEDATA_TEST_URL = "https://172.16.238.222"
+WEBMON_TEST_URL = "http://localhost"
+
+
+class TestPlotlyVersions:
+    user = "InstrumentScientist"
+    pwd = "InstrumentScientist"
+
+    @classmethod
+    def setup_class(cls):
+        """Clean the database and create test data"""
+        conn = psycopg2.connect(
+            database=os.environ.get("DATABASE_NAME", "workflow"),
+            user=os.environ.get("DATABASE_USER", "workflow"),
+            password=os.environ.get("DATABASE_PASS", "workflow"),
+            port=os.environ.get("DATABASE_PORT", 5432),
+            host=os.environ.get("DATABASE_HOST", "localhost"),
+        )
+        cur = conn.cursor()
+
+        # Clean up previous test data
+        cur.execute("DELETE FROM plots_plotdata")
+        cur.execute("DELETE FROM plots_datarun")
+        cur.execute("DELETE FROM plots_instrument")
+        conn.commit()
+
+        # Create test data for ARCS run 214583
+        add_instrument_data_run(conn, "arcs", "IPTS-27800", 214583)
+
+        # Create test data for REF_L run 299096
+        add_instrument_data_run(conn, "ref_l", "IPTS-33077", 299096)
+
+        conn.close()
+
+    def get_session(self):
+        URL = WEBMON_TEST_URL + "/users/login"
+        client = requests.session()
+
+        # Retrieve the CSRF token first
+        client.get(URL)  # sets the cookie
+        csrftoken = client.cookies["csrftoken"]
+
+        login_data = dict(username="InstrumentScientist", password="InstrumentScientist", csrfmiddlewaretoken=csrftoken)
+        response = client.post(URL, data=login_data)
+        assert response.status_code == 200
+        return client
+
+    def test_publish_versions(self, db_connection, request_page):
+        """Test that plots are published with the correct plotlyjs-version attributes."""
+
+        # Trigger reduction for ARCS (using autoreducer with Plotly v5)
+        run_number_arcs = 214583
+        response = request_page(f"/report/arcs/{run_number_arcs}/reduce/", self.user, self.pwd)
+        assert response.status_code == 200, f"ARCS reduction request failed: {response.status_code}"
+
+        # Wait for processing
+        time.sleep(3)
+
+        # Check that ARCS plot has plotlyjs-version="5"
+        key = generate_key("arcs", run_number_arcs)
+        plot_url = f"{LIVEDATA_TEST_URL}/plots/arcs/{run_number_arcs}/update/html/?key={key}"
+        plot_response = requests.get(plot_url, verify=False)
+        assert plot_response.status_code == 200, f"Failed to fetch ARCS plot: {plot_response.status_code}"
+        assert "ARCS Test Plot - Run 214583" in plot_response.text
+        assert 'plotlyjs-version="5.17.0' in plot_response.text, "ARCS plot should have plotlyjs-version='5'"
```

Comment:
So `plot_publisher` is currently giving us the wrong version. It's giving us the version of the python plotly library (`plotly.__version__`) where we need the `plotlyjs` version.  It should instead be doing `from plotly.offline import get_plotlyjs_version; get_plotlyjs_version()`

In `detail.html` it is correctly checking the `plotlyjs-version` string `startsWith('3.')`

But once `plot_publisher` is updated we will need to change update these tests.